From 93df0656c8ae15d5b731c8ade7ce3483a7bd0730 Mon Sep 17 00:00:00 2001
From: Michal Suchanek <msuchanek@suse.de>
Date: Fri, 19 Jan 2018 19:11:32 +0100
Subject: [PATCH] Revert "cxl: Fix driver use count"

References: bsc#1066223
Patch-mainline: v4.9-rc2
Git-commit: 70b565bbdb911023373e035225ab10077e4ab937

Revert the stable commit so it can be reapplied in-order.
Keep the original stable commit so series2git can patch the tree on top of
stable tree.
---
 drivers/misc/cxl/api.c  | 4 ----
 drivers/misc/cxl/file.c | 8 +-------
 2 files changed, 1 insertion(+), 11 deletions(-)

diff --git a/drivers/misc/cxl/api.c b/drivers/misc/cxl/api.c
index 690eb1a18caf..ea3eeb7011e1 100644
--- a/drivers/misc/cxl/api.c
+++ b/drivers/misc/cxl/api.c
@@ -176,10 +176,6 @@ int cxl_start_context(struct cxl_context *ctx, u64 wed,
 		kernel = false;
 	}
 
-	/*
-	 * Increment driver use count. Enables global TLBIs for hash
-	 * and callbacks to handle the segment table
-	 */
 	cxl_ctx_get();
 
 	if ((rc = cxl_attach_process(ctx, kernel, wed , 0))) {
diff --git a/drivers/misc/cxl/file.c b/drivers/misc/cxl/file.c
index 013558f4da4f..10a02934bfc0 100644
--- a/drivers/misc/cxl/file.c
+++ b/drivers/misc/cxl/file.c
@@ -94,6 +94,7 @@ static int __afu_open(struct inode *inode, struct file *file, bool master)
 
 	pr_devel("afu_open pe: %i\n", ctx->pe);
 	file->private_data = ctx;
+	cxl_ctx_get();
 
 	/* indicate success */
 	rc = 0;
@@ -204,18 +205,11 @@ static long afu_ioctl_start_work(struct cxl_context *ctx,
 	ctx->pid = get_task_pid(current, PIDTYPE_PID);
 	ctx->glpid = get_task_pid(current->group_leader, PIDTYPE_PID);
 
-	/*
-	 * Increment driver use count. Enables global TLBIs for hash
-	 * and callbacks to handle the segment table
-	 */
-	cxl_ctx_get();
-
 	trace_cxl_attach(ctx, work.work_element_descriptor, work.num_interrupts, amr);
 
 	if ((rc = cxl_attach_process(ctx, false, work.work_element_descriptor,
 				     amr))) {
 		afu_release_irqs(ctx, ctx);
-		cxl_ctx_put();
 		goto out;
 	}
 
-- 
2.13.6

